providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -60,24 +61,41 @@ typedef struct ml_kem_gen_ctx_st {
     uint8_t *seed;
 } PROV_ML_KEM_GEN_CTX;
 
-#ifdef FIPS_MODULE
 static int ml_kem_pairwise_test(const ML_KEM_KEY *key)
 {
-    int ret = 0;
+#ifdef FIPS_MODULE
     OSSL_SELF_TEST *st = NULL;
     OSSL_CALLBACK *cb = NULL;
     void *cbarg = NULL;
+#endif
     unsigned char secret[ML_KEM_SHARED_SECRET_BYTES];
     unsigned char out[ML_KEM_SHARED_SECRET_BYTES];
     unsigned char entropy[ML_KEM_RANDOM_BYTES];
     unsigned char *ctext = NULL;
     const ML_KEM_VINFO *v = ossl_ml_kem_key_vinfo(key);
+    EVP_MD_CTX *mdctx;
     int operation_result = 0;
+    int ret = 0;
 
     /* Unless we have both a public and private key, we can't do the test */
-    if (!ossl_ml_kem_have_prvkey(key))
+    if (!ossl_ml_kem_have_prvkey(key) || !ossl_ml_kem_have_pubkey(key))
         return 1;
 
+    /*
+     * Hash of private key storage (sans "d", which may not be set) is the
+     * encapsulation entropy source.
+     */
+    if ((mdctx = EVP_MD_CTX_new()) == NULL
```

Comment:
@paulidale Care to comment on this bit?

Reply:
This isn't useful.  Just use a buffer of zeros so things aren't slowed down any.

Reply:
Now just using default (random) entropy. Simpler.